tools/src/versioning/android/versionVendoredModules.ts

```diff
@@ -0,0 +1,227 @@
+import spawnAsync from '@expo/spawn-async';
+import assert from 'assert';
+import chalk from 'chalk';
+import fs from 'fs-extra';
+import glob from 'glob-promise';
+import path from 'path';
+
+import { ANDROID_DIR } from '../../Constants';
+import logger from '../../Logger';
+import { copyFileWithTransformsAsync } from '../../Transforms';
+import { FileTransforms } from '../../Transforms.types';
+import { searchFilesAsync } from '../../Utils';
+import vendoredModulesTransforms from './transforms/vendoredModulesTransforms';
+
+const ANDROID_VENDORED_DIR = path.join(ANDROID_DIR, 'vendored');
+
+/**
+ * Versions Android vendored modules.
+ */
+export async function versionVendoredModulesAsync(
+  sdkNumber: number,
+  filterModules: string[] | null
+): Promise<void> {
+  const prefix = `abi${sdkNumber}_0_0`;
+  const config = vendoredModulesTransforms(prefix);
+  const baseTransforms = baseTransformsFactory(prefix);
+  const unversionedDir = path.join(ANDROID_VENDORED_DIR, 'unversioned');
+  const versionedDir = vendoredDirectoryForSDK(sdkNumber);
+  let vendoredModuleNames = await getVendoredModuleNamesAsync(unversionedDir);
+  if (filterModules) {
+    vendoredModuleNames = vendoredModuleNames.filter((name) => filterModules.includes(name));
+  }
+
+  for (const name of vendoredModuleNames) {
+    logger.info('🔃 Versioning vendored module %s', chalk.green(name));
+
+    const moduleConfig = config[name];
+    const sourceDirectory = path.join(unversionedDir, name);
+    const targetDirectory = path.join(versionedDir, name);
+    const files = await searchFilesAsync(sourceDirectory, '**');
+
+    await fs.remove(targetDirectory);
+
+    for (const sourceFile of files) {
+      await copyFileWithTransformsAsync({
+        sourceFile,
+        sourceDirectory,
+        targetDirectory,
+        transforms: {
+          path: [...baseTransforms.path, ...(moduleConfig?.path ?? [])],
+          content: [...baseTransforms.content, ...(moduleConfig?.content ?? [])],
+        },
+      });
+    }
+
+    await maybePrebuildSharedLibsAsync(name, sdkNumber);
+  }
+}
+
+/**
+ * Prebuild shared libraries to jniLibs and cleanup CMakeLists.txt
+ */
+async function maybePrebuildSharedLibsAsync(module: string, sdkNumber: number) {
+  const moduleRootDir = path.join(ANDROID_DIR, 'vendored', `sdk${sdkNumber}`, module, 'android');
```

Comment:
why are you adding new type of identifier for SDK here, everything on android is using `abix_0_0`